src/runtime/rocm/rocm_device_api.cc

```diff
@@ -0,0 +1,143 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ * \file rocm_device_api.cc
+ * \brief GPU specific API
+ */
+#include <tvm/runtime/config.h>
+#include <tvm/runtime/device_api.h>
+
+#if TVM_ROCM_RUNTIME
+#include <dmlc/logging.h>
+#include <dmlc/thread_local.h>
+#include <tvm/runtime/registry.h>
+#include <hip/hip_runtime_api.h>
+#include <hsa/hsa.h>
+#include "./rocm_common.h"
+
+namespace tvm {
+namespace runtime {
+
+class ROCMDeviceAPI final : public DeviceAPI {
+ public:
+  void SetDevice(TVMContext ctx) final {
+    ROCM_CALL(hipSetDevice(ctx.device_id));
+  }
+  void GetAttr(TVMContext ctx, DeviceAttrKind kind, TVMRetValue* rv) final {
+    int value = 0;
+    switch (kind) {
+      case kExist:
+        value = (hsa_init() == HSA_STATUS_SUCCESS);
```

Comment:
- Do we need to include hsa library into LD_LIBRARY_PATH?
- This do not yet check if the specific device_id exist. For example, i can try rocm(1).exist, but I only have gpu 0.